Simple changes to the top-level requirements

AdamuKaapan opened the pull request:
- Added keeping track of position at all times to manual mode as well
 - Added monitoring for Mode-Switch messages to manual mode as well

This commit is to make the top-level requirements a bit more complete,
adding things that I think were missed in my opinion. The position
tracking I added because it'll make transitioning back into autonomous
easier (which the ability to do is in the requirements already). The
mode-switch monitoring in manual is mainly just to make it consistent
between autonomous and manual mode.

(this commit is also just to give me an excuse to test using this style of Git workflow)

AdamuKaapan committed:
Simple changes to the top-level requirements

 - Added keeping track of position at all times to manual mode as well
 - Added monitoring for Mode-Switch messages to manual mode as well

This commit is to make the top-level requirements a bit more complete,
adding things that I think were missed in my opinion. The position
tracking I added because it'll make transitioning back into autonomous
easier (which the ability to do is in the requirements already). The
mode-switch monitoring in manual is mainly just to make it consistent
between autonomous and manual mode.

Jebbs commented on Top-Level Requirements.md:
```diff
@@ -25,14 +25,16 @@ Top-Level System Requirements
 * While in Manual Mode, the Robot **shall**:
   * Monitor network traffic for controller information
+  * At all times, keep track of its position
+  * Monitor network traffic for a “Mode-Switch” command
```
Since this is listed in both, why not put it in the "While in either mode" section?

collinsc replied:
I agree with jeremy about moving the wording of keep track of position, but I would phrase it as maintain localization, that is the language we have used for the rest of requirements.

I think keeping the language as "Monitor network traffic for controller information" is more prudent. What if we wanted to transmit more kinds of commands than "Mode Switch"? I am not saying get rid of it. I am saying let's keep the more general wording as top level, and make several specific sub requirements.

AdamuKaapan replied:
I agree with all 3 points of those, lemme change that

AdamuKaapan replied:
I actually didn't even see the either mode part before, so that's useful.

collinsc reviewed (changes requested):
Clean up the wording and placement a little. See comments.

AdamuKaapan committed:
Updated to fit pull request review

 - Moved some points to "both" section
 - Rephrased "position" to "localization" for consistency
 - Rephrased "mode-switch command" to "controller commands" to maintain
   a more top-level approach
